File: clothesshop-frontend-app/src/services/ProductService.js
import axios from "axios";

const BASE_API = "http://localhost:8080/api/products";

class ProductService {

    getProducts() {
        return axios.get(BASE_API + "/all");
    }

    insertProduct(product) {
        return axios.post(BASE_API + "/insert",product);
    }

    getProduct(id){
        return axios.get(BASE_API+ "/find/"+id)
    }
    
    updateProduct(id,product){
        return axios.put(BASE_API+"/update/"+id,product)
    }

    deleteProduct(id){
        return axios.delete(BASE_API+"/delete/"+id)
    }

}

export default new ProductService();

File: Spring Boot/src/main/java/wi/pb/clothesshop/service/ProductService.java
package wi.pb.clothesshop.service;

import org.springframework.stereotype.Service;
import wi.pb.clothesshop.entity.Product;

import java.util.List;


public interface ProductService {

    List<Product> getAllProducts();

    Product insertProduct(Product product);

    Product getProduct(int id);

    Product updateProduct(int id, Product product);

    void deleteProduct(int id);

}
